sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java

```diff
@@ -310,85 +311,105 @@ private synchronized void receiveMessages() {
         }
         ServiceBusReceiverAsyncClient receiverClient = asyncClient.get();
 
+        ReceiverOptions receiverOptions = receiverClient.getReceiverOptions();
+        // If it is session processor, we create subscribers for each session.
+        if (receiverOptions.isRollingSessionReceiver()) {
+            receiverClient.receiveMessagesWithContextFromRollingSessions()
+                .subscribe(this::subscribeReceiveFlux);
```

Comment:
After testing, here I need to handle errors here, like connection inactive case. I'll add onError/onComplete hook to restart client like messageSubscriber does.